lib/PuppeteerSharp/ElementHandle.cs

```diff
@@ -629,20 +633,26 @@ public async Task DragAndDropAsync(ElementHandle target, int delay = 0)
             }
 
             await ScrollIntoViewIfNeededAsync().ConfigureAwait(false);
-            var (x, y) = await ClickablePointAsync().ConfigureAwait(false);
+            var clickablePoint = await ClickablePointAsync().ConfigureAwait(false);
             var targetPoint = await target.ClickablePointAsync().ConfigureAwait(false);
-            await Page.Mouse.DragAndDropAsync(x, y, targetPoint.X, targetPoint.Y, delay).ConfigureAwait(false);
+            await Page.Mouse.DragAndDropAsync(clickablePoint.X, clickablePoint.Y, targetPoint.X, targetPoint.Y, delay).ConfigureAwait(false);
         }
 
-        private async Task<(decimal X, decimal Y)> ClickablePointAsync()
+        /// <summary>
+        /// Returns the middle point within an element unless a specific offset is provided.
+        /// </summary>
+        /// <param name="offset">Optional offset</param>
+        /// <exception cref="PuppeteerException">When the node is not visible or not an HTMLElement</exception>
+        /// <returns>A <see cref="Task"/> that resolves to the clickable point</returns>
+        public async Task<BoxModelPoint> ClickablePointAsync(BoxModelPoint? offset = null)
```

Comment:
Rather than using Boxmodelpoint for the offset I'd like to propose a new Offset struct be created/used. Matching the current puppeteer implementation. The Offset class in puppeteer has specific descriptions for the x/y properties. 

https://pptr.dev/api/puppeteer.offset

The struct is largely self documenting then.

Reply:
Disagree :)

Reply:
Puppeteer has an offset object so puppeteer Sharp should have one too 

> x-offset for the clickable point relative to the top-left corder of the border box.

The properties are documented differently to Boxmodelpoint